src/test/java/htmlpublisher/HtmlFileNameTest.java

```diff
@@ -24,21 +24,21 @@ public void fileNameWithSpecialCharacters() throws Exception {
 
         FreeStyleProject job = j.createFreeStyleProject();
 
-        job.getBuildersList().add(new CreateFileBuilder("subdir/#$&+,;= @.html", content));
+        job.getBuildersList().add(new CreateFileBuilder("subdir/subdir2/#$&+,;= @.html", content));
```

Comment:
maybe you could add another test to test this subdirectory path as well?

Reply:
Sorry, I'm not familiar with the APIs that are used throughout this test, and I'm also not set up to run the test suite locally, so I didn't want to take the risk of inadvertently breaking something.

At the expense of duplicating a substantial amount of test code, I now went ahead and introduced a second test instead of trying to cover both cases in a single function. Hope that's okay. I'm sure someone who's more knowledgeable about this code would be able to provide a much cleaner implementation.

Reply:
perfect!!